backend/de.metas.business/src/main/java/de/metas/project/workorder/project/WOProjectRepository.java

```diff
@@ -154,10 +156,98 @@ public List<WOProject> listBy(@NonNull final WOProjectQuery query)
 	{
 		return computeQuery(query)
 				.stream()
-				.map(WorkOrderProjectRepository::ofRecord)
+				.map(WOProjectRepository::ofRecord)
 				.collect(ImmutableList.toImmutableList());
 	}
 
+	@NonNull
+	public ImmutableSet<ProjectId> getActiveProjectIds()
+	{
+		return queryBL
+				.createQueryBuilder(I_C_Project.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_C_Project.COLUMNNAME_ProjectCategory, ProjectCategory.WorkOrderJob)
+				.create()
+				.listIds(ProjectId::ofRepoId);
+	}
+
+	@NonNull
+	public List<WOProject> getAllActiveProjectsByProjectCalendarQuery(@NonNull final WOProjectCalendarQuery query)
+	{
+		return query.isNone()
+				? ImmutableList.of()
+				: computeProjectCalendarQuery(query)
+				.stream()
+				.map(WOProjectRepository::ofRecord)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	@NonNull
+	public ImmutableSet<ProjectId> getActiveProjectIdsByProjectCalendarQuery(@NonNull final WOProjectCalendarQuery query)
+	{
+		return getAllActiveProjectsByProjectCalendarQuery(query)
+				.stream()
+				.map(WOProject::getProjectId)
+				.collect(ImmutableSet.toImmutableSet());
+	}
+
+	@Nullable
+	private I_C_Project getRecordById(@NonNull final ProjectId id)
+	{
+		return InterfaceWrapperHelper.load(id, I_C_Project.class);
+	}
+
+	@NonNull
+	private IQuery<I_C_Project> computeQuery(@NonNull final WOProjectQuery query)
+	{
+		final IQueryBuilder<I_C_Project> queryBuilder = queryBL.createQueryBuilder(I_C_Project.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_C_Project.COLUMNNAME_ProjectCategory, ProjectCategory.WorkOrderJob.getCode())
+				.addInArrayFilter(I_C_Project.COLUMNNAME_AD_Org_ID, query.getOrgId(), OrgId.ANY)
+				.orderByDescending(I_C_Project.COLUMNNAME_AD_Org_ID);
+
+		if (EmptyUtil.isNotBlank(query.getExternalProjectReferencePattern()))
+		{
+			queryBuilder.addStringLikeFilter(I_C_Project.COLUMNNAME_C_Project_Reference_Ext, "%" + query.getExternalProjectReferencePattern() + "%", true);
+		}
+
+		if (query.getExternalId() != null)
+		{
+			queryBuilder.addEqualsFilter(I_C_Project.COLUMNNAME_ExternalId, query.getExternalId().getValue());
+		}
+
+		if (EmptyUtil.isNotBlank(query.getValue()))
+		{
+			queryBuilder.addEqualsFilter(I_C_Project.COLUMNNAME_Value, query.getValue().trim());
+		}
+
+		return queryBuilder.create();
+	}
+
+	@NonNull
+	private IQuery<I_C_Project> computeProjectCalendarQuery(@NonNull final WOProjectCalendarQuery query)
+	{
+		final IQueryBuilder<I_C_Project> queryBuilder = queryBL
+				.createQueryBuilder(I_C_Project.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_C_Project.COLUMNNAME_ProjectCategory, ProjectCategory.WorkOrderJob)
```

Comment:
i checked org.adempiere.ad.dao.impl.CompareQueryFilter#toString and i don't think that is will "unbox" WorkOrderJob's code in a reliable manner, so AFAIU you need to use `ProjectCategory.WorkOrderJob.getCode()`, no?

PS: if you know it does, please augment the javadoc of de.metas.util.lang.ReferenceListAwareEnum accordingly

Reply:
I've just tested this method and the unboxing worked fine (also checked org.adempiere.ad.dao.impl.CompareQueryFilter#accept and when the values are compared you can see that  org.adempiere.ad.dao.impl.CompareQueryFilter#normalizeValue method covers also the `ReferenceListAwareEnum` case )